### PR TITLE
feat(core): add socialConnectors details for get sign-in-settings

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add social connector details for `GET /sign-in-settings` API.

## Linear Issue Reference

LOG-2389

## Testing

UTs.

### CODE REVIEW

File: packages/ui/src/hooks/use-social.ts [hunk 1]
```diff
@@ -90,10 +90,8 @@ const useSocial = () => {
   // Filter native supported social connectors
   const socialConnectors = useMemo(
     () =>
-      (experienceSettings?.socialConnectors ?? []).filter(({ target }) => {
-        return (
-          !isNativeWebview() || getLogtoNativeSdk()?.supportedSocialConnectorIds.includes(target)
-        );
+      (experienceSettings?.socialConnectors ?? []).filter(({ id }) => {
```
Comment (demonzoo) [inline packages/ui/src/hooks/use-social.ts:93]: nit: `experienceSettings?.socialConnectors?.filter`

**Review by demonzoo — APPROVED** [review]
Overall looks good to me.

File: packages/core/src/routes/session.ts [hunk 2]
```diff
@@ -577,9 +577,15 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
...
     const signInExperience = await findDefaultSignInExperience();
-    ctx.body = signInExperience;
+    const connectorInstances = await getConnectorInstances();
+    const instanceMap = new Map(
+      connectorInstances.map((instance) => [instance.connector.id, instance])
+    );
+    const socialConnectors = signInExperience.socialSignInConnectorIds.map((id) => {
+      return { ...instanceMap.get(id)?.metadata, id };
+    });
+    ctx.body = { ...signInExperience, socialConnectors };
```
Comment (wangsijie) [inline packages/core/src/routes/session.ts:588]: `socialConnector` have methods? better remove them or pick data manualy.

Comment (darcyYe) [reply packages/core/src/routes/session.ts:588]: it only contains metadata and connector.id

Comment (wangsijie) [reply packages/core/src/routes/session.ts:588]: 看走眼了

**Review by wangsijie — APPROVED** [review]

**Pull request merged**